lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/batch/AzureBatchHelper.java

```diff
@@ -41,27 +44,44 @@
  */
 public final class AzureBatchHelper {
 
-  private static final Logger LOG = Logger.getLogger(AzureBatchJobSubmissionHandler.class.getName());
+  private static final Logger LOG = Logger.getLogger(AzureBatchHelper.class.getName());
 
   /*
    * Environment variable that contains the Azure Batch jobId.
    */
   private static final String AZ_BATCH_JOB_ID_ENV = "AZ_BATCH_JOB_ID";
+  private static final String CAPTURE_HOST_IP_ADDRESS_TASK_NAME = "CaptureHostIpAddress";
 
   private final AzureBatchFileNames azureBatchFileNames;
 
   private final BatchClient client;
   private final PoolInformation poolInfo;
+  private final CommandBuilder commandBuilder;
+  private final ContainerRegistryProvider containerRegistryProvider;
+  private final boolean areContainersEnabled;
+  private TcpPortProvider portProvider;
 
   @Inject
   public AzureBatchHelper(
       final AzureBatchFileNames azureBatchFileNames,
       final IAzureBatchCredentialProvider credentialProvider,
+      final TcpPortProvider portProvider,
+      final CommandBuilder commandBuilder,
```

Comment:
Should we be using AzureBatchPoolDriverPortsList to stay inline with AzureBatchService.cs?

Reply:
I cannot use AzureBatchPoolDriverPortsList in the java side since it is a List class that is not avro serializable. Hence the use of the set instead on the java side.

---
In reply to: [201793020](https://github.com/sharathmalladi/reef/pull/5#discussion_r201793020) [](ancestors = 201793020)